components/Input/DateInput/index.js

```diff
@@ -96,7 +97,7 @@ const decodeDate = (value, separator) => {
 
 // value is string
 const isValidDateString = (value, separator) => {
-    if (value === '' || value === undefined) {
+    if (isFalsy(value, [''])) {
```

Comment:
We only allow empty string and `undefined` as valid value. So `isFalsy` isn't required here.